core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts

```diff
@@ -82,14 +98,39 @@ export class DotEditPageToolbarComponent implements OnInit, OnChanges, OnDestroy
                 page: {
                     pageState: this.pageState,
                     pageRenderedHtml: this.pageRenderedHtml || null
+                },
+                onSave: () => {
+                    this.setDotFavoritePageHighlighted();
+                    this.loadFavoritePageData();
                 }
             }
         });
     }
 
+    /**
+     * Calls ES endpoint to verify if DotFavoritePage contentlet exist on current page
+     *
+     * @memberof DotEditPageToolbarComponent
+     */
+    loadFavoritePageData(): void {
+        const urlParam = generateDotFavoritePageUrl(this.pageState);
+
+        this.dotPageStateService
+            .requestFavoritePageData(urlParam)
+            .subscribe((response: ESContent) => {
+                if (response.resultsSize > 0) {
+                    this.setDotFavoritePageHighlighted();
+                }
+            });
+    }
+
     private updateRenderedHtml(): string {
         return this.pageState?.params.viewAs.mode === DotPageMode.PREVIEW
             ? this.pageState.params.page.rendered
             : this.pageRenderedHtml;
     }
+
+    private setDotFavoritePageHighlighted(): void {
+        this.dotFavoritePageIconName = 'grade';
```

Comment:
why grade ? what it means ?

Reply:
I think it is this [icon](https://materialui.co/icon/grade)